keras/preprocessing/sequence.py

```diff
@@ -216,3 +218,93 @@ def _remove_long_seq(maxlen, seq, label):
             new_seq.append(x)
             new_label.append(y)
     return new_seq, new_label
+
+
+class TimeSeriesSequence(Sequence):
+    """Utility class for generating batches of temporal data.
+
+    This class takes in a sequence of data-points gathered at 
+    equal intervals, along with time series parameters such as 
+    stride, delay etc., to produce batches for training/validation.
+
+    # Arguments
+        data: indexable generator (such as list) containing
+            consecutive data points.
+        targets: targets corresponding to instances in `data`.
+            Should have same length as data.
+        lookback: length of history to consider for prediction.
+            data[i-lookback : i+1] is used as input.
+        delay: how further in the future we are looking to predict.
+            targets[i + delay] becomes the target for
+            data[i-lookback : i+1]
+        sampling_rate: period between successive individual data
+            points. For rate `r`, points
+            data[i], data[i-r], ... data[i - lookback] are used as input.
+        stride: period between successive sub-sequences used for
+            training. For stride s, consecutive training samples would
+            be centered around data[i], data[i+s], data[i+2*s], ...
+        start_index, end_index: delimiters for sub-sets of data & targets
+            to be used for batch generation. Useful to reserve part of
+            data for validation.
+        shuffle: whether to shuffle training samples, or draw them in
+            chronological order.
+        reverse: if true, input points are provided in reverse order.
+        batch_size: size of successful batches (except maybe last one)
```

Comment:
There are different "sizes" at play here (number of samples, number of timesteps...). Please specify "number of samples (timeseries) in a batch"